public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -544,3 +546,18 @@ function isPromQuery(model: AlertDataQuery): model is PromQuery {
 export function isPromOrLokiQuery(model: AlertDataQuery): model is PromOrLokiQuery {
   return 'expr' in model;
 }
+
+// the backend will always execute "hidden" queries, so we have no choice but to remove the property in the front-end
```

Comment:
I move these two functions to avoid duplication in the two forms, as i think it worths in this case to not duplicate.